ci/integTestGen/src/integTestGen.jl

```diff
@@ -0,0 +1,265 @@
+to analyze as dependency.
+
+# Args
+    `project_path::AbstractString`: Absolute path to the project folder of the package to analyse
+"""
+function create_working_env(project_path::AbstractString)
+    tmp_path = mktempdir()
+    Pkg.activate(tmp_path)
+    # same dependency like in the Project.toml of integTestGen.jl
+    Pkg.add("Pkg")
+    Pkg.add("PkgDependency")
+    Pkg.add("YAML")
+    # add main project as dependency
+    Pkg.develop(path=project_path)
+end
+
+"""
+Contains all information about a package.
+
+# Fields
+- `url`: Git url of the original project.
+- `modified_url`: Stores the Git url set by the environment variable.
+- `env_var`: Name of the environment variable to set the modified_url.
+"""
+mutable struct PackageInfo
+    url::String
+    modified_url::String
+    env_var::String
+    PackageInfo(url, env_var) = new(url, "", env_var)
+end
+
+package_infos = Dict(
+    "QED" => PackageInfo(
+        "https://github.com/QEDjl-project/QED.jl.git",
+        "CI_INTG_PKG_URL_QED"),
+    "QEDfields" => PackageInfo(
+            "https://github.com/QEDjl-project/QEDfields.jl.git",
+            "CI_INTG_PKG_URL_QEDfields"),
+    "QEDbase" => PackageInfo(
+            "https://github.com/QEDjl-project/QEDbase.jl.git",
+            "CI_INTG_PKG_URL_QEDbase"),
+    "QEDevents" => PackageInfo(
+            "https://github.com/QEDjl-project/QEDevents.jl.git",
+            "CI_INTG_PKG_URL_QEDevents"),
+    "QEDprocesses" => PackageInfo(
+            "https://github.com/QEDjl-project/QEDprocesses.jl.git",
+            "CI_INTG_PKG_URL_QEDprocesses"),
+)
+
+"""
+    extract_env_vars_from_git_message!()
+
+    Parse the commit message, if set via variable (usual `CI_COMMIT_MESSAGE`) and set custom URLs.
+"""
+function extract_env_vars_from_git_message!(var_name = "CI_COMMIT_MESSAGE")
+    if haskey(ENV, var_name)
+        for line in split(ENV[var_name], "\n")
+            line = strip(line)
+            for pkg_info in values(package_infos)
+                if startswith(line, pkg_info.env_var * ": ")
+                    ENV[pkg_info.env_var] = SubString(line, length(pkg_info.env_var * ": ") + 1)
+                end
+            end
+        end
+    end
+end
+
+"""
+    set_modified_package_url!()
+
+Iterate over all entries of package_info. If an environment variable exits with the same name like,
+the `env_var` entry, set the value of the environment variable to `custom_url`.
+"""
+function set_modified_package_url!()
+    for package_info in values(package_infos)
+        if haskey(ENV, package_info.env_var)
+            package_info.modified_url = ENV[package_info.env_var]
+        end
+    end
+end
+
+"""
+    get_modified_package()::String
+
+Read the name of the modified (project) package from the environment variable `CI_DEPENDENCY_NAME`.
+
+# Returns
+- The name of the modified (project) package
+"""
+function get_modified_package()::String
```

Comment:
is it really necessary to cast the output of this function into a `String`?

Reply:
Actual ENV[] returns a `String` (e.g. `typeof(ENV["USER"])`). I thought, it is a good idea to use the type for documentation purpose of the function.

Reply:
We clarified it offline. I removed the `String`